crates/block-multiplier/benches/bench.rs

```diff
@@ -0,0 +1,75 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::prelude::StdRng;
+use rand::{Rng, SeedableRng};
+
+fn bench_block_multiplier(c: &mut Criterion) {
+    let mut group = c.benchmark_group("block_multiplier");
+
+    let seed: u64 = rand::random();
+    println!("Using random seed for benchmark: {}", seed);
+    let mut rng = StdRng::seed_from_u64(seed);
+
+    let s0_a = [
+        rng.random::<u64>(),
+        rng.random::<u64>(),
+        rng.random::<u64>(),
+        rng.random::<u64>(),
+    ];
```

Comment:
These can be written more concisely as: 
```suggestion
    let s0_a = std::array::from_fn(|_| rng.random::<u64>())
```
and let the compiler infer how many entries it should have.